### DISCUSSION TITLE
Best faders? Encoders and analouge?

### DISCUSSION THREAD

**Discussion by 2fourk** [shanteacontrols/OpenDeck#202] [Q&A] [answered]
Hi, I am in middle of selecting components to do my first proper build. 

I am overwhelmed by fader options. I was wondering if there was a good guide anywhere? Or even if someone could make some suggestions.

**Comment by fdisp** [comment, marked as answer]
hi @2fourk I am actually suffering the same problem myself.
after a few weeks researching for parts (momentary push buttons, rotary potentiometers, sliding potentiometers, encoders, and their respective datasheets)
to minimize my problems and the need of designing another pcb required for surface mount family of components,
I excluded any smt type, so we are left only with panel mount parts, which only require work (cuts and drills) on the front panel.
along the way I found out each component has some negative properties which need a mitigation strategy.
one of those negative properties are called "bouncing". it is present in push buttons when the button switches state for a few milliseconds and in contacting encoders in the edge of the switching pulses. (bouncing is not present in magnetic and optical encoders but these are more expensive parts and also need extra work and electronic parts to adapt in the circuit).
the negative property affecting potentiometers (both sliding and rotating) is noise. while the middle pin slides along the conductive film it makes mechanical noise which in turn is translated to electrical noise.
I found myself forced to design an extra pcb which eliminate them to the extent that makes the used parts reliable. the pcb which will mitigate the negative effects is a work in progress and will be shared when it is finished.
as for the components I liked the specifications of the parts made by bourns, specially their long life (100k cycles) PTF/PTE line of sliding potentiometers. the PTL line which have an integrated LED looked cool but on second thought, longer life and reliability is more important than good looking. also the bourns rotary potentiometers had good specifications but they are expensive.
https://bourns.com/products/potentiometers
https://bourns.com/products/encoders

after realizing the euro distributors (specially uk based) would not ship the parts to a 3-rd world country like the one I'm living in ,
I decided to go for alpsalpine by Chinese distributors
https://tech.alpsalpine.com/e/products/category/potentiometer/
https://tech.alpsalpine.com/e/products/category/encorder/

the push buttons and switches are another whole new story of research. I'm still currently researching them as they come with all sort of different shapes, force/interaction-behavior and bounce specifications. I came to realize that for the sake of good user experience and optimal performance the buttons need to be soft and depending on the bouncing specification in the datasheet , a proper de-bouncing mitigation interface is required. the simplest/cheapest form being a capacitor in parallel and the best/more expensive, using de-bouncing IC's for this very purpose, like for ex MC14490
these along with contacting encoders require an extra pcb where bouncing, for all the switches, push buttons or contacting encoders is filtered. it is not hard to design but one is required to go along the process of design, picking extra components like capacitors, resistors, ic's  like SN74LVC1G17 or MC14490 pinheads/wire mounting parts and fabrication.
this extra pcb will serve as an intermediate interface between opendeck and the physical controllers in the front panel.

there exists another form of bouncing mitigation which is done through the firmware/software running in the microcontroller/cpu.
I have not heard anything about that in the opendeck firmware. @paradajz is the one who can answer this.

**Reply by paradajz** [reply]
> I have not heard anything about that in the opendeck firmware. @paradajz is the one who can answer this.

Button debouncing: https://github.com/shanteacontrols/OpenDeck/blob/master/src/application/io/buttons/Filter.h
Encoder debouncing: https://github.com/shanteacontrols/OpenDeck/blob/master/src/application/io/encoders/Filter.h
Analog debouncing: https://github.com/shanteacontrols/OpenDeck/blob/master/src/application/io/analog/Filter.h

**Reply by 2fourk** [reply]
Thank you so much!!! So much useful information

**Comment by paradajz** [comment]
> Hi, I am in middle of selecting components to do my first proper build.
> 
> I am overwhelmed by fader options. I was wondering if there was a good guide anywhere? Or even if someone could make some suggestions.

Did you check Mouser? There is also this wiki page:
https://github.com/shanteacontrols/OpenDeck/wiki/Recommended-components-and-where-to-get-them

**Reply by 2fourk** [reply]
I have brought a few things from mouser over the years

**Comment by 2fourk** [comment]
> Hi, I am in middle of selecting components to do my first proper build.
> 
> I am overwhelmed by fader options. I was wondering if there was a good guide anywhere? Or even if someone could make some suggestions.

I have discovered this linear encoder strip 
![Screenshot_20230503_122643_Chrome](https://user-images.githubusercontent.com/5184233/235823562-97a1b284-a200-4a5e-96d3-50150847595a.jpg)

Which seems like it could be a cheap option for creating cheap linear encoders. I have asked some engineering friends for help in understanding.

**Comment by 2fourk** [comment]
Also I brought this cheap $60 midi controller to hook up as my prototype and test, I was thinking depending on the size of the dmx electronics I might even just keep them in and add a switch for the open deck midi controls. Also means that it's still useful when I move open deck to full prototype.


![-1281246220477905515](https://user-images.githubusercontent.com/5184233/235823796-f9ffdc31-0fc5-4815-8017-26e9d4162422.jpg)
